### PR TITLE
Emi's branch

### PR DESCRIPTION
**Pull request by Emily-Case**
Added data.c and all of it's functions

### CODE REVIEW

File: src/data.c [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    return 0;
+}
+
+/* Função que duplica uma estrutura data_t, reservando a memória
+ * necessária para a nova estrutura.
+ * Retorna a nova estrutura ou NULL em caso de erro.
+ */
+struct data_t *data_dup(struct data_t *data){
+
+    if(data == NULL || data->data == NULL || data->datasize <= 0)
```
Comment (padrezulmiro) [inline src/data.c:43]: Just as a comment, you can leave it as it is, but it's debatable whether data->data has to be different from NULL.

File: src/data.c [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+ * Deve assegurar que liberta o espaço ocupado pelo conteúdo antigo.
+ * Retorna 0 (OK) ou -1 em caso de erro.
+ */
+int data_replace(struct data_t *data, int new_size, void *new_data){
+
+    if(data == NULL || data->data == NULL || data->datasize <= 0 || new_size <= 0 || new_data == NULL)
+        return -1;
+
+    free(data->data);
+    free(data->datasize);
```
Comment (padrezulmiro) [inline src/data.c:63]: As far as I understand, this shouldn't work. You might've not found an issue bc your intellisense is not working yet, or bc **everything** in C is technically a number, including pointers. But data->datasize is not a pointer as required by `free()`. It's actually an `int`.

Comment (padrezulmiro) [reply src/data.c:63]: image

**Review by padrezulmiro — CHANGES_REQUESTED** [review]
I left a few notes for the issues that need fixing.

File: src/data.c (hunk 1 above)
Comment (padrezulmiro) [reply src/data.c:43]: Again, there's nothing to correct, just a comment

File: src/data.c (hunk 2 above)
Comment (Emily-Case) [reply src/data.c:63]: ah ofc, that was just dumb of me, yes it does not give an error, but I was aware of it, I simply forgor

File: src/data.c [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    d->datasize = size;
+
+    return d;
+}
+
+/* Função que elimina um bloco de dados, apontado pelo parâmetro data,
+ * libertando toda a memória por ele ocupada.
+ * Retorna 0 (OK) ou -1 em caso de erro.
+ */
+int data_destroy(struct data_t *data){
```
[the remark this answers is not in the document]
Comment (enderesting) [reply src/data.c:27]: I didn't quite get this one, wouldn't free (data) do it? it is a pointer to the struct after all

[the remark this answers is not in the document]
Comment (Emily-Case) [reply src/data.c:27]: will be fixed :)

**Review by enderesting — APPROVED** [review]
all good to me!

**Pull request merged**